sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/CryptographyClient.cs

```diff
@@ -0,0 +1,346 @@
+        }
+
+
+        /// <summary>
+        /// Signs the specified digest.
+        /// </summary>
+        /// <param name="digest">The digest to sign</param>
+        /// <param name="algorithm">The algorithm to use</param>
+        /// <param name="cancellationToken">Cancellation token</param>
+        public virtual async Task<SignResult> SignAsync(SignatureAlgorithm algorithm, byte[] digest, CancellationToken cancellationToken = default)
+        {
+            return await _cryptoProvider.SignAsync(algorithm, digest, cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Signs the specified digest.
+        /// </summary>
+        /// <param name="digest">The digest to sign</param>
+        /// <param name="algorithm">The algorithm to use</param>
+        /// <param name="cancellationToken">Cancellation token</param>
+        public virtual SignResult Sign(SignatureAlgorithm algorithm, byte[] digest, CancellationToken cancellationToken = default)
+        {
+            return _cryptoProvider.Sign(algorithm, digest, cancellationToken);
+        }
+
+        /// <summary>
+        /// Verifies the specified signature value
+        /// </summary>
+        /// <param name="digest">The digest</param>
+        /// <param name="signature">The signature value</param>
+        /// <param name="algorithm">The algorithm to use</param>
+        /// <param name="cancellationToken">Cancellation token</param>
+        public virtual async Task<VerifyResult> VerifyAsync(SignatureAlgorithm algorithm, byte[] digest, byte[] signature, CancellationToken cancellationToken = default)
+        {
+            return await _cryptoProvider.VerifyAsync(algorithm, digest, signature, cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Verifies the specified signature value
+        /// </summary>
+        /// <param name="digest">The digest</param>
+        /// <param name="signature">The signature value</param>
+        /// <param name="algorithm">The algorithm to use</param>
+        /// <param name="cancellationToken">Cancellation token</param>
+        public virtual VerifyResult Verify(SignatureAlgorithm algorithm, byte[] digest, byte[] signature, CancellationToken cancellationToken = default)
+        {
+            return _cryptoProvider.Verify(algorithm, digest, signature, cancellationToken);
+        }
+
+
+        /// <summary>
+        /// Signs the specified data
+        /// </summary>
+        /// <param name="data">The data to sign</param>
+        /// <param name="algorithm">The algorithm to use</param>
+        /// <param name="cancellationToken">Cancellation token</param>
+        public virtual async Task<SignResult> SignDataAsync(SignatureAlgorithm algorithm, byte[] data, CancellationToken cancellationToken = default)
+        {
+            if (data == null) throw new ArgumentNullException(nameof(data));
```

Comment:
I only check data because this is the only input which is referenced by the client to compute its hash. All other inputs are passed directly to the service.